### DISCUSSION TITLE
Best way to share immutable data between components?

### DISCUSSION THREAD

**Discussion by arun-mani-j** [nuxt/nuxt#15991] [Questions] [answered]
Hello all.
In my website, there is a navigation menu, which is displayed in header when the screen size is large and shown as a sidebar for users on mobile.
I'm using daisyUI to create this [setup](https://daisyui.com/components/drawer/).
However, there is a duplication of data between the header and sidebar, and I want to avoid it.

My menu structure is a simple JSON.
```json
[
  {
    "name": "About Us",
    "to": "/about"
  },
...
]
```

I found composables and especially `useState` to share mutable data. But since the data here is immutable, I don't think they deserve the semantics of a _state_. 
Another method is to use `useFetch` and store the JSON file inside `public` folder and fetch it to build the menu.
Last method is to store the JSON in some folder like `data` and read it via NodeJS modules like `fs`.

I'm pretty confused which way to go, any suggestions are highly welcome!

Thanks for reading. :)

**Comment by kanopa** [comment, marked as answer]
Why not just create a file with constants and put your array there?

**Reply by arun-mani-j** [reply]
Oops, I completely forgot the existence of import/export functionality in JS/TS :see_no_evil: 
Thanks!
